src/Coordinates/Vector2.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Vector2.hpp"
+#include <cmath>
+
+template<class T>
+Vector2<T>::Vector2(T x, T y) :
+	x(x),
+	y(y)
+{}
+
+template<class T>
+Vector2<T> Vector2<T>::abs() const {
+	return Vector2<T>(std::abs(this->x), std::abs(this->y));
+}
+
+template<class T>
+Vector2<T> Vector2<T>::operator+(Vector2<T> other) const {
+	return Vector2<T>(this->x + other.x, this->y + other.y);
+}
+
+template<class T>
+Vector2<T> Vector2<T>::operator-(Vector2<T> other) const {
```

Comment:
nit: I think the operators should be passed vector by const ref, i.e. `const Vector2<T>& other` to prevent them making a copy. https://en.cppreference.com/w/cpp/language/operators

Reply:
I think Eigen should be used as it does delayed evaluation otherwise we are looking at a hit in speed hence why I coded up #213

Reply:
1) I believed the slowdown (when I was writing that) to be negligable, a Vector2 is just two doubles.

2) Despite that, I switched it to a const ref later on. I don't have any performance metrics on that, though.